File: 2020/02/02-1.go
package main

import (
	"fmt"
	"strconv"
	"strings"

	. "../helpers"
)

func main() {
	var lines []string
	var err error
	if lines, err = ReadInput("./input"); err == nil {
		fmt.Printf("something bad", err)
	}

	validPasswords := 0

	for _, line := range lines {
		split_line := strings.Split(line, " ")

		charRange := split_line[0]
		keyChar := []rune(strings.Replace(split_line[1], ":", "", -1))
		password := strings.Trim(split_line[2], " \n")

		splitCharRange := strings.Split(charRange, "-")
		minChars, _ := strconv.Atoi(splitCharRange[0])
		maxChars, _ := strconv.Atoi(splitCharRange[1])

		count := 0
		for _, char := range password {
			if char == keyChar[0] {
				count++
			}
		}

		if count >= minChars && count <= maxChars {
			validPasswords++
		}
	}
	fmt.Printf("Valid Passwords: %d", validPasswords)
}

File: 02/02-1.py
from collections import Counter
from sys import exit

if __name__ == "__main__":    
    valid = 0
    with open('input', 'r') as o:
        for line in o.readlines():
            char_range, char, password = line.split(" ")

            char = char[0] # remove :
            password = password.strip() # remove newline
            
            min_chars, max_chars = [int(x) for x in char_range.split("-")]

            c = Counter(password)

            if c[char] >= min_chars and c[char] <= max_chars:
                valid += 1
    
    print("Valid passwords: ", valid)

